### ISSUE TITLE
Test Failure: SearchableSnapshotActionIT.testConvertingPartialSearchableSnapshotIntoFull

### ISSUE THREAD

**Issue by jakelandis** [elastic/elasticsearch#69396]
**Build scan**:

https://gradle-enterprise.elastic.co/s/oap2ugwldcgrq

**Repro line**:
```
./gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testConvertingPartialSearchableSnapshotIntoFull" -Dtests.seed=DCEA435C28CB3E70 -Dtests.security.manager=true -Dtests.locale=tr -Dtests.timezone=BET -Druntime.java=8
```
**Reproduces locally?**:

no (tried against 5a730d4bce41983ddf2362e1734b88402ce5d835 ) 

**Applicable branches**:

* master
* 7.x
* 7.12

**Failure history**:

https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Chicago&tests.container=org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT&tests.sortField=FAILED&tests.test=testConvertingPartialSearchableSnapshotIntoFull&tests.unstableOnly=true

53 failures in past 4 days. 
61 in past 2 weeks. 
Seems to have started on Feb. 9. 

![image](https://user-images.githubusercontent.com/976291/108781098-ff790a00-752e-11eb-8b3e-91c216ea6604.png)

**Failure excerpt**:

The build shows a few failed tests. But pretty sure the one from the repo line is the offender. The others failed due to the server crashed. The server appears to have crashed due to an exception on the cluster state applier thread:

```
» [2021-02-22T21:56:50,001][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [javaRestTest-3] fatal error in thread [elasticsearch[javaRestTest-3][clusterApplierService#updateTask][T#1]], exiting	
»  java.lang.AssertionError: null	
»  	at org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:427) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  	at org.elasticsearch.cluster.service.ClusterApplierService.access$000(ClusterApplierService.java:57) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  	at org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:151) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:241) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:204) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_281]	
»  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_281]	
»  	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_281]
```

Which seems to be caused by : 
```
»  	Suppressed: java.lang.AssertionError: current thread [Thread[elasticsearch[javaRestTest-3][clusterApplierService#updateTask][T#1],5,main]] may not read [name: __jXMFlCJWSByh90KY89DTRA, numberOfParts: 1, partSize: 8192pb, partBytes: 9223372036854775807, metadata: name [_0.cfe], length [405], checksum [teje4i], writtenBy [8.8.0]]	
»  		at org.elasticsearch.index.store.BaseSearchableSnapshotIndexInput.assertCurrentThreadMayAccessBlobStore(BaseSearchableSnapshotIndexInput.java:249) ~[?:?]	
»  		at org.elasticsearch.index.store.BaseSearchableSnapshotIndexInput.openInputStreamFromBlobStore(BaseSearchableSnapshotIndexInput.java:128) ~[?:?]	
»  		at org.elasticsearch.index.store.cache.FrozenIndexInput.readDirectlyIfAlreadyClosed(FrozenIndexInput.java:424) ~[?:?]	
»  		at org.elasticsearch.index.store.cache.FrozenIndexInput.doReadInternal(FrozenIndexInput.java:391) ~[?:?]	
»  		at org.elasticsearch.index.store.BaseSearchableSnapshotIndexInput.readInternal(BaseSearchableSnapshotIndexInput.java:83) ~[?:?]	
»  		at org.apache.lucene.store.BufferedIndexInput.refill(BufferedIndexInput.java:315) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.store.BufferedIndexInput.readByte(BufferedIndexInput.java:56) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.store.BufferedChecksumIndexInput.readByte(BufferedChecksumIndexInput.java:41) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.store.DataInput.readInt(DataInput.java:102) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.codecs.CodecUtil.checkHeader(CodecUtil.java:194) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.codecs.CodecUtil.checkIndexHeader(CodecUtil.java:255) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.codecs.lucene50.Lucene50CompoundReader.readEntries(Lucene50CompoundReader.java:107) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:08:18]	
»  		at org.apache.lucene.codecs.lucene50.Lucene50CompoundReader.<init>(Lucene50CompoundReader.java:68) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:08:18]	
»  		at org.apache.lucene.codecs.lucene50.Lucene50CompoundFormat.getCompoundReader(Lucene50CompoundFormat.java:71) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:08:18]	
»  		at org.apache.lucene.index.SegmentCoreReaders.<init>(SegmentCoreReaders.java:101) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.index.SegmentReader.<init>(SegmentReader.java:83) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:66) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:58) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.index.SegmentInfos$FindSegmentsFile.run(SegmentInfos.java:720) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.index.StandardDirectoryReader.open(StandardDirectoryReader.java:81) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.apache.lucene.index.DirectoryReader.open(DirectoryReader.java:63) ~[lucene-core-8.8.0.jar:8.8.0 b10659f0fc18b58b90929cfdadde94544d202c4a - noble - 2021-01-25 19:07:45]	
»  		at org.elasticsearch.index.engine.ReadOnlyEngine.openDirectory(ReadOnlyEngine.java:548) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.index.engine.FrozenEngine.getOrOpenReader(FrozenEngine.java:158) ~[?:?]	
»  		at org.elasticsearch.index.engine.FrozenEngine.openSearcher(FrozenEngine.java:231) ~[?:?]	
»  		at org.elasticsearch.index.engine.FrozenEngine.access$000(FrozenEngine.java:44) ~[?:?]	
»  		at org.elasticsearch.index.engine.FrozenEngine$2.acquireSearcherInternal(FrozenEngine.java:192) ~[?:?]	
»  		at org.elasticsearch.index.engine.Engine$SearcherSupplier.acquireSearcher(Engine.java:1225) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.index.engine.Engine.acquireSearcher(Engine.java:677) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.index.engine.Engine.acquireSearcher(Engine.java:670) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.index.engine.Engine.acquireSearcher(Engine.java:666) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.index.engine.ReadOnlyEngine.getRawFieldRange(ReadOnlyEngine.java:567) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.index.shard.IndexShard.getTimestampRange(IndexShard.java:1808) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.indices.cluster.IndicesClusterStateService.updateShard(IndicesClusterStateService.java:615) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.indices.cluster.IndicesClusterStateService.createOrUpdateShards(IndicesClusterStateService.java:544) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:230) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:499) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:489) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:460) ~[elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:407) [elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.cluster.service.ClusterApplierService.access$000(ClusterApplierService.java:57) [elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:151) [elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:673) [elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:241) [elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:204) [elasticsearch-7.13.0-SNAPSHOT.jar:7.13.0-SNAPSHOT]	
»  		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_281]	
»  		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_281]	
»  		at java.lang.Thread.run(Thread.java:748) [?:1.8.0_281]
```

Full stack trace in [build scan's console log](https://gradle-enterprise.elastic.co/s/oap2ugwldcgrq/console-log#L12819).

**Comment by jakelandis** [comment]
Muted on 
* master aa83a8e29692e4432912149f842a2fe34df5e5f2
* 7.x 683fb11096d06916b01a5060b5c0bc5e854e889e
* 7.12 d7e018d7d41dfe12120c6f22908cb63180e9295b

**Comment by jakelandis** [comment]
might be related to https://github.com/elastic/elasticsearch/issues/56233 (similar error signature)

**Comment by jakelandis** [comment]
@ywelsch -  is there a related fix for these tests ?  Also, it appears that this was reverted on 7.x and 7.12, but not master ?

**Comment by ywelsch** [comment]
> is there a related fix for these tests ?

The error shown above is something that was fixed by #69357.

> Also, it appears that this was reverted on 7.x and 7.12, but not master ?

It was reverted on master as well, see 6305fc6a151?

**Comment by jakelandis** [comment]
thanks @ywelsch !

> It was reverted on master as well, see 6305fc6?

ahh.. i just need to learn how to read GH's UI better :)
